components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java

```diff
@@ -172,6 +170,24 @@ public OMElement readAndCleanWsdl2(API api) throws APIManagementException {
         }
     }
 
+    /**
+     * Gets WSDL definition as a byte array given the WSDL definition
+     * @param wsdlDefinition generated WSDL definition
+     * @return converted WSDL definition as byte array
+     * @throws APIManagementException
+     */
+    public byte[] getWSDL(Definition wsdlDefinition) throws APIManagementException {
+        try {
+            WSDLWriter writer = getWsdlFactoryInstance().newWSDLWriter();
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            writer.writeWSDL(wsdlDefinition, byteArrayOutputStream);
+            return byteArrayOutputStream.toByteArray();
+        } catch (Exception e) {
+            String msg = "Error occurs when change the address URL of the WSDL";
+            throw new APIManagementException(msg, e);

```

Comment:
Here you can  just put the error message as in bellow
 throw new APIManagementException("Error occurs when change the address URL of the WSDL", e);